### PR TITLE
Add Base8 Spec

### PR DESCRIPTION
**Pull request by fabianhjr**
This is a first throw at specifying a Base8 encoding, solves some issues of #59 and multiformats/go-multibase#26

Currently the test vectors mapped to the octal value of each byte so encoding and decoding would be harder than need be and encoded bytearray values would be larger (if each byte is mapped to 3 chars) or unpredictable (if, as currently, there are some chars encoded to different octal lengths)

For example, the first test vector was:

(In chunks of 3)
`["104","312","615","453","347","216","230","266","151","364","624","403","127","314","534","474","564","320","645","563","162","044","1"]`

(In chunks of 8)
`["10431261","54533472","16230266","15136462","44031273","14534474","56432064","55631620","441"]`

And `312` seems to be the encoding of `e` sometimes in some places like `position ≡ {1,2} mod 3` .

### CODE REVIEW

File: rfcs/Base8.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+## Encoding
+
+Map each 3 byte, with the most significant bit on the left side as follows:
+
+`[⁰b₇₆₅₄₃₂₁₀, ¹b₇₆₅₄₃₂₁₀, ²b₇₆₅₄₃₂₁₀]`
+
+`[⁰b₇₆₅ ⁰b₄₃₂ ⁰b₁₀¹b₇ ¹b₆₅₄ ¹b₃₂₁ ¹b₀²b₇₆ ²b₅₄₃ ²b₂₁₀]`
+
+Treat missing subbits as zero (as in `⁰b₁₀¹b₇` if `⁰b` is defined and `¹b₀²b₇₆`
+if `¹b₀` is defined) and optionally pad with `=`
```
Comment (Stebalien) [inline rfcs/Base8.md]: Can we simplify this a bit?

File: rfcs/Base8.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Base8
+
+The multibase base8 prefix is the character `7`. Canonical multibase-base8
+encoded data will always have `length ≡ {1, 4, 7} mod 8` if unpadded or
+`length = 8k + 1` for some `k ∈ ℕ` if padded with `=`.
```
[the remark this answers is not in the document]
Comment (ribasushi) [reply rfcs/Base8.md]: @fabianhjr I agree with @Stebalien : let's remove the `*pad` parts from the test vectors entirely.

[the remark this answers is not in the document]
Comment (fabianhjr) [reply rfcs/Base8.md]: @ribasushi, test vectores removed.

**Comment by ribasushi** [comment]
@hugomrdias when time permits can you check that the spec-as-written makes sense? You were the last to implement stuff in this area...

**Comment by fabianhjr** [comment]
I think it would be best to wait until the encoding is a go before trying to spec the decoding. As far as I know this encoding would fit what was implemented in multibase-js however will wait for @hugomrdias.

**Comment by fabianhjr** [comment]
Thanks @hugomrdias, @ribasushi added the decoding section.

File: rfcs/Base8.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+Then map each 8 3-bit words, with the most significant bit on the left side
+(big-endian), to 3 bytes (8-bit word) as follows:
+
+`[⁰b₂₁₀, ¹b₂₁₀, ²b₂₁₀, ³b₂₁₀, ⁴b₂₁₀, ⁵b₂₁₀, ⁶b₂₁₀, ⁷b₂₁₀]`
+
+`[⁰b₂₁₀¹b₂₁₀²b₂₁, ²b₀³b₂₁₀⁴b₂₁₀⁵b₂, ⁵b₁₀⁶b₂₁₀⁷b₂₁₀]`
+
+If there are not enough bits to complete an 8-bit word then drop those last
+bits.
```
Comment (hugomrdias) [inline rfcs/Base8.md:45]: Does this mean this https://github.com/multiformats/js-multibase/blob/61af3471d1382db29aee4ba396532cf5226a2e46/src/rfc4648.js#L41-L44 is too strict and should be removed ?

Comment (fabianhjr) [reply rfcs/Base8.md]: Sorry, not familiar with Javascript to understand the nuances however if it can decode the first 2 test cases it should be fine.

**Comment by fabianhjr** [comment]
Hi @Stebalien, I think this is ready to be merged. UwU

**Comment by Stebalien** [comment]
:rocket:

**Pull request merged**